cmd/kubeadm/app/cmd/alpha/certs_test.go

```diff
@@ -285,3 +294,212 @@ func TestRenewUsingCSR(t *testing.T) {
 		t.Fatalf("couldn't load certificate %q: %v", cert.Name, err)
 	}
 }
+
+// A minimal kubeadm config with just enough values to avoid triggering
+// auto-detection of config values at runtime.
+const kubeadmConfig = `
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: InitConfiguration
+localAPIEndpoint:
+  advertiseAddress: 192.0.2.1
+nodeRegistration:
+  criSocket: /path/to/dockershim.sock
+---
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: ClusterConfiguration
+certificatesDir: /custom/config/certificates-dir
+kubernetesVersion: v1.19.0
+`
+
+func TestGenCSR(t *testing.T) {
```

Comment:
This looks more like an integration test to me. Possibly, we need to refactor and put it in `cmd/kubeadm/test/cmd`.